makefile

```diff
@@ -70,5 +70,16 @@ getdependency:
 #######################################
 # compile all tests under $(TEST_ROOT)
 #######################################
+COMPILATION_LOG=$(TESTOUTPUT)$(D)compile$(D)compilation.log
+
+MOVE_TDUMP_PERL=perl scripts$(D)moveDmp.pl --compileLogPath=$(Q)$(COMPILATION_LOG)$(Q)
+MOVE_TDUMP=if [ -z $$(tail -n 1 $(COMPILATION_LOG) | grep 0) ]; then $(MOVE_TDUMP_PERL); $(RM) $(Q)$(COMPILATION_LOG)$(Q); else $(RM) $(Q)$(COMPILATION_LOG)$(Q); fi
+COMPILE_CMD=ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJDK_VERSION=$(JDK_VERSION) -DJDK_IMPL=$(JDK_IMPL) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DPLATFORM=${PLATFORM} -DTEST_JDK_HOME=${TEST_JDK_HOME} -DJVM_VERSION=$(JVM_VERSION)
+
 compile: getdependency
-	ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJDK_VERSION=$(JDK_VERSION) -DJDK_IMPL=$(JDK_IMPL) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DPLATFORM=${PLATFORM} -DTEST_JDK_HOME=${TEST_JDK_HOME} -DJVM_VERSION=$(JVM_VERSION)
+ifneq (,$(findstring zos,$(SPEC)))
+	$(MKTREE) $(TESTOUTPUT)$(D)compile;
+	($(COMPILE_CMD) 2>&1; echo $$? ) | tee $(Q)$(COMPILATION_LOG)$(Q); $(MOVE_TDUMP)
+else
+	$(COMPILE_CMD)
+endif
```

Comment:
Talked with @JennyChenyj and @renfeiw , we also need to capture cores for other platforms. Please remove the if in this make file. For simplicity, the logic should be in `moveDmp.pl`.

Reply:
Due to the time constraint, we will deliver the zos support first.